packages/plugin-drizzle/src/helper.ts

```diff
@@ -0,0 +1,224 @@
+import {} from "drizzle-orm/pg-core";
+import {
+  type PgliteDatabase as OriginalPgliteDatabase,
+  drizzle as drizzlePGLite,
+} from "drizzle-orm/pglite";
+import { migrate as migratePGLite } from "drizzle-orm/pglite/migrator";
+import pg from "pg";
+
+/**
+ * Union type of all possible drizzle database options
+ */
+export type DrizzleOptions = PgliteDrizzleOptions | NodePgDrizzleOptions;
+
+/**
+ * Configuration options for Node-Postgres database connection
+ */
+export type NodePgDrizzleOptions = {
+  /**
+   * Type of database to use -
+   * - "pglite" - PGLite database
+   * - "node-postgres" - Node-Postgres database
+   * @default "pglite"
+   */
+  type: "node-postgres";
+  /**
+   * Connection string to use for the database
+   * @default ""
+   */
+  connectionString?: string;
+  /**
+   * Pool configuration options for Node-Postgres
+   */
+  poolConfig?: pg.PoolConfig;
+  /**
+   * Additional drizzle configuration options
+   */
+  config?: Omit<DrizzleConfig, "schema">;
+};
+
+/**
+ * Configuration options for PGLite database connection
+ */
+export type PgliteDrizzleOptions = {
+  /**
+   * Type of database to use -
+   * - "pglite" - PGLite database
+   * - "node-postgres" - Node-Postgres database
+   */
+  type?: "pglite";
+  /**
+   * Connection string to use for the database
+   * @default "memory://pglite"
+   */
+  connectionString?: string;
+  /**
+   * Pool configuration is not supported for PGLite
+   */
+  poolConfig?: never;
+  /**
+   * Additional drizzle configuration options with PGLite specific connection options
+   */
+  config?: Omit<DrizzleConfig, "schema"> & {
+    connection?:
+      | (PGliteOptions & {
+          dataDir?: string;
+        })
+      | string;
+  };
+};
+
+/**
+ * Extended PGLite database type with client information
+ */
+export type PgliteDatabase<TSchema extends Record<string, unknown>> =
+  OriginalPgliteDatabase<TSchema> & {
+    $client: PGlite;
+  };
+
+/**
+ * Extended Node-Postgres database type with client information
+ */
+export type NodePgDatabase<TSchema extends Record<string, unknown>> =
+  OriginalNodePgDatabase<TSchema> & {
+    $client: pg.Pool;
+  };
+
+export type Database<
+  TOptions extends DrizzleOptions,
+  TSchema extends Record<string, unknown>,
+> = TOptions extends PgliteDrizzleOptions
+  ? PgliteDatabase<TSchema>
+  : NodePgDatabase<TSchema>;
+
+/**
+ * Creates a new Drizzle database instance based on the provided options
+ * @param options - Configuration options for the database connection
+ * @returns A configured Drizzle database instance
+ * @throws {Error} If an invalid database type is specified
+ */
+export function drizzle<
+  TSchema extends Record<string, unknown>,
+  TOptions extends DrizzleOptions,
+>(
+  options?: TOptions & {
+    /**
+     * Schema to use for the database
+     * @default {}
+     */
+    schema?: TSchema;
+  },
+): Database<TOptions, TSchema> {
+  const {
+    connectionString = "memory://",
+    schema,
+    type = "pglite",
```

Comment:
IMHO this default together with the check below makes the API confusing.

The API should work like the following:

* type unspecified: accept only connection string as parameter. If connectionString is undefined, use in-memory db (and in this case print a warning).
* type specified: accept type specific options like now.